api/src/main/java/org/apache/iceberg/RewriteJobOrder.java

```diff
@@ -0,0 +1,62 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.iceberg;
+
+import java.util.Locale;
+import org.apache.iceberg.relocated.com.google.common.base.Preconditions;
+
+/**
+ * Enum of supported rewrite job order, it defines the order in which the file groups should be
+ * written.
+ * <p>
+ * 1. bytes: rewrite the smallest job groups first.
+ * <p>
+ * 2. files: rewrite the job groups with the most files first.
+ * <p>
+ * 3. none: rewrite job groups in the order they were planned (no specific ordering).
+ */
+public enum RewriteJobOrder {
+  BYTES("bytes"), FILES("files"), NONE("none");
+
+  private final String orderName;
+
+  RewriteJobOrder(String orderName) {
+    this.orderName = orderName;
+  }
+
+  public String orderName() {
+    return orderName;
+  }
+
+  public static RewriteJobOrder fromName(String orderName) {
+    Preconditions.checkNotNull(orderName, "Name of rewrite job order should not be null");
+    return RewriteJobOrder.valueOf(orderName.toUpperCase(Locale.ENGLISH));
+  }
+
+  public static boolean isValid(String value) {
+    for (RewriteJobOrder rewriteJobOrder : RewriteJobOrder.values()) {
```

Comment:
I'm not sure you need this, if youd did it would probably be sufficent to see if 
"fromName" throws an exception.

RewriteJobOrder.valueOf() will throw a   ```if the specified enum type has no constant with the specified name, or the specified class object does not represent an enum type.```

In general we just call `fromName` and let the illegal arg exception bubble up.

If you wanted to make an isValid method specifically (I don't think you need one here) you could just call 
```java
try {
  fromName(value)
  return true
} catch {
  return false
}

Reply:
Yes, it should be possible using `fromName`. I just want to validate value of `rewriteJobOrder` in `validateAndInitOptions()`.